backend/corpora/dataset_processing/process.py

```diff
@@ -548,19 +546,18 @@ def get_bucket_prefix(dataset_id):
 
 
 def process_cxg(local_filename, dataset_id, cellxgene_bucket):
-    bucket_prefix = get_bucket_prefix(dataset_id)
     cxg_dir = convert_file_ignore_exceptions(make_cxg, local_filename, "Issue creating cxg.", dataset_id, "cxg_status")
     if cxg_dir:
-        update_db(dataset_id, processing_status={"cxg_status": ConversionStatus.UPLOADING})
-        s3_uri = copy_cxg_files_to_cxg_bucket(cxg_dir, bucket_prefix, cellxgene_bucket)
-        metadata = {
-            "explorer_url": join(
-                DEPLOYMENT_STAGE_TO_URL[os.environ["DEPLOYMENT_STAGE"]],
-                dataset_id + ".cxg",
-                "",
-            )
-        }
         with db_session_manager() as session:
+            # generate an s3_uri
+            while True:
+                cxg_id = generate_uuid()
+                bucket_prefix = get_bucket_prefix(cxg_id)
+                s3_uri = f"s3://{cellxgene_bucket}/{bucket_prefix}.cxg/"
+                # check if the s3_uri is in the asset table
+                filters = [DatasetAsset.table.s3_uri == s3_uri]
+                if session.query(DatasetAsset.table).filter(*filters).first():
+                    break
```

Comment:
Isn't this searching for a _newly generated_ UUID in the db and then breaking out of the infinite loop only if it exists? Is my understanding of the logic check inverted? Sorry if I'm being dense...

Reply:
you're right, I gotta invert it.

Reply:
😅 I stared at this for a while doubting myself, so glad I caught something in the end!

If this is trying to handle the case was a UUID might be reused, I'm not sure it's worth the defensive code tbh. Probability is squarely on our side here. And if the db is checking uniqueness, we have that defense as well.

Reply:
the db doesn't check for the uniqueness of the s3_uri so it would pass through

Reply:
I modified the code to use the asset id rather than generate a new UUID. This will give us the DB uniqueness check.